### DISCUSSION TITLE
raspberry pi picoひとつだけでcargo runしたい

### DISCUSSION THREAD

**Discussion by kk0kojirr** [KOBA789/rusty-keys#3] [Q&A]
raspberry pi picoひとつだけ接続してcargo runした時にprobeがなく、ファームウェアが書き込みできないようでありますが、書き込みはprobeのpico経由でされているのでしょうか。

**Comment by KOBA789** [comment]
その通りです！
書き込みにはデバッグアダプタ(probe)が必要です。
uf2 などを用いて単独で書き込む方法もありますが、それだけではまともなデバッグができないためこのようにしています。
https://rusty-keys.koba789.com/firmware/connection

**Reply by ciniml** [reply]
デバッガ使わずに書き込みたいのであればelf2uf2-rs使えばできると思いますけど、たぶんブートモードに手動で入れたりしないといけないので面倒な気がします。
